### PR TITLE
[Tw 127] remove compiler warnings

### PR DESCRIPTION
**Pull request by kantp**
This reduces the noise at compile time, making it easier to spot warnings that might be important.

Also, replaced some incomplete pattern matches with Exceptions.

### CODE REVIEW

File: src/Node/Internal.hs [hunk 1]
```diff
@@ -412,7 +413,7 @@ stAddHandler !provenance !statistics = case provenance of
             , stRunningHandlersLocalAverage = runningHandlersLocalAverage
             }
 
-    Remote !peer _ _ -> do
+    Remote _ _ _ -> do
```
Comment (avieth) [inline src/Node/Internal.hs]: Should leave the bang `!_`

Comment (avieth) [inline src/Node/Internal.hs]: And on line 400 as well.

File: src/Node/Internal.hs [hunk 2]
```diff
@@ -773,10 +774,10 @@ nodeDispatcher node handlerIn handlerInOut =
     connectionOpened
         :: DispatcherState peerData m
         -> NT.ConnectionId
-        -> NT.Reliability
+        -> NT.Reliability -- TODO: this is not actually used. If this is intentional, we should remove the parameter.
```
Comment (avieth) [inline src/Node/Internal.hs]: You can go ahead and remove it if you like. It's certainly not used now, we'll put it back if we ever need it.

**Comment by avieth** [comment]
LGTM after those bang patterns are reinserted. Thanks @kantp

File: src/Node/Internal.hs (hunk 1 above)
Comment (kantp) [inline src/Node/Internal.hs]: @avieth: also on lines 464 and 481?

Comment (avieth) [inline src/Node/Internal.hs]: Ah yes, good catch.

**Pull request merged**